recipes/libcurl/all/conanfile.py

```diff
@@ -347,7 +347,7 @@ def _patch_mingw_files(self):
     def _build_with_autotools(self):
         with tools.chdir(self._source_subfolder):
             # autoreconf
-            self.run("./buildconf", win_bash=tools.os_info.is_windows, run_environment=True)
+            self.run("autoreconf -fiv", win_bash=tools.os_info.is_windows, run_environment=True)
```

Comment:
```suggestion
            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows, run_environment=True)
```

Reply:
this will raise if AUTORECONF not in the env

Reply:
Might need to require libtool for mingw also in build_requirement.

Reply:
if AUTORECONF is in env since we have autoconf in build requirement (through libtool)

Reply:
build_requirement also needs to be updated, to not break MinGW:

```python
    def build_requirements(self):
        if not self._is_using_cmake_build:
            self.build_requires("libtool/2.4.6")
            self.build_requires("pkgconf/1.7.3")
            if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
                self.build_requires("msys2/20200517")
        elif self._is_win_x_android:
            self.build_requires("ninja/1.10.2")
```

Reply:
I've just checked MinGW, and it fails for me:
```
libcurl/7.74.0: run_in_windows_bash: C:\.conan\d42a89\1\usr\bin\bash.exe --login -c ^"cd \^"/c/users/sse4/.conan/data/libcurl/7.74.0/_/_/build/33db161173eeaea75b0b6758cf47c091b549bebd/source_subfolder\^" ^&^& PATH=\^"/c/users/sse4/.conan/data/cmake/3.16.4/_/_/package/456f15897172eef340fcbac8a70811f2beb26a93/bin:/c/users/sse4/.conan/data/mingw_installer/1.0/conan/stable/package/d6e9d8771a7904278203ca1b715b751651fb5664/bin:/c/.conan/d42a89/1/usr/bin:/c/.conan/c5d4a7/1/bin/usr/bin:$PATH\^" ^&^& None -fiv ^"                                                                                   /usr/bin/bash: None: command not found                            
```
seems like solution to use `AUTORECONF` is not right

Reply:
You mean it fails with the modification suggested above in build requirement, or without?

Reply:
without
I am trying with now, but it will need to build the world (m4 and so on)

Reply:
fails with as well:
```
libcurl/7.74.0: run_in_windows_bash: C:\.conan\d42a89\1\usr\bin\bash.exe --login -c ^"cd \^"/c/users/sse4/.conan/data/libcurl/7.74.0/_/_/build/33db161173eeaea75b0b6758cf47c091b549bebd/source_subfolder\^" ^&^& PATH=\^"/c/users/sse4/.conan/data/cmake/3.16.4/_/_/package/456f15897172eef340fcbac8a70811f2beb26a93/bin:/c/users/sse4/.conan/data/mingw_installer/1.0/conan/stable/package/d6e9d8771a7904278203ca1b715b751651fb5664/bin:/c/.conan/d42a89/1/usr/bin:/c/users/sse4/.conan/data/libtool/2.4.6/_/_/package/e99faf16acec7281f7fd81c5696eb4133d3bc92e/bin:/c/users/sse4/.conan/data/pkgconf/1.7.3/_/_/package/6dc82da13f94df549e60f9c1ce4c5d11285a4dff/bin:/c/.conan/c5d4a7/1/bin/usr/bin:/c/users/sse4/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin:/c/users/sse4/.conan/data/autoconf/2.69/_/_/package/3475bd55b91ae904ac96fde0f106a136ab951a5e/bin:/c/users/sse4/.conan/data/m4/1.4.18/_/_/package/413a343c621037ce4d92e0bfc6f082ecc5ba0001/bin:$PATH\^" ^&^& /c/users/sse4/.conan/data/autoconf/2.69/_/_/package/3475bd55b91ae904ac96fde0f106a136ab951a5e/bin/autoreconf.exe -fiv ^"                                                                                     autoreconf.exe: Entering directory `.'                                                                                                              autoreconf.exe: configure.ac: not using Gettext                                                                                                     autoreconf.exe: running: /c/users/sse4/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/aclocal.exe --force -I m4                                                                                                                                                   aclocal.exe: error: couldn't open directory '/c/users/sse4/.conan/data/libtool/2.4.6/_/_/package/e99faf16acec7281f7fd81c5696eb4133d3bc92e/bin/share/aclocal;/c/users/sse4/.conan/data/pkgconf/1.7.3/_/_/package/6dc82da13f94df549e60f9c1ce4c5d11285a4dff/bin/aclocal': No such file or directory        autoreconf.exe: /c/users/sse4/.conan/data/automake/1.16.3/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/aclocal.exe failed with exit status: 1                                                                                                                                               
```

Reply:
okay, it works **WITHOUT** `libtool/2.4.6` and `pkgconf/1.7.3` in build requirements, with the following fix:
```
self.run("{} -fiv".format(tools.get_env("AUTORECONF") or "autoreconf"), win_bash=tools.os_info.is_windows, run_environment=True)
```
it turns out, MinGW has `autoreconf` which works out of the box, so it doesn't need build requirement (?)
should I commit these changes then? 
I am not sure, MinGW support goes outside of the scope of the PR. 
so, probably just leave it as it was before (calling `./buildconf`)?

Reply:
it's fine for me (keep buildconf, and replace this logic in another PR).

Just one comment, and @madebr may confirm: msys2 recipe does not guarantee to provide autoreconf of pkg-config.

Reply:
@a4z I probably will revert back to `buildconf` for now, as it seems to be problematic to avoid breakage for MinGW. we may try one more time in another PR :)

Reply:
See https://github.com/curl/curl/blob/ec5d9b44a2e837fc7b82d1c60d5fae3f851620dc/buildconf#L3
I don't see the point in using buildconf

Reply:
as I understand, `buildconf` script is more or less equivalent to:
```
self.run("{} -fiv".format(tools.get_env("AUTORECONF") or "autoreconf"), win_bash=tools.os_info.is_windows, run_environment=True)
```
so, this change should be okay

Reply:
What do you need autoreconf for here anyway? Wouldn't configure be enough?